### ISSUE TITLE
[BUG] Java referenced condition not finding an expected match

### ISSUE THREAD

**Issue by pranavgaikwad** [konveyor/analyzer-lsp#378]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

beta.1

### Priority

Important

### Current Behavior

The rule [azure-file-system-01000](https://github.com/konveyor/rulesets/blob/975a3e56195875bcf937c48c572c717a8412f68c/default/generated/azure/01-azure-file-system.windup.yaml#L1-L34) should be matched. 

### Expected Behavior

When running against [book-server](https://github.com/ibraginsky/book-server) app, following condition should be matched:

```
  when:
    java.referenced:
      pattern: java.io.(File|FileWriter|FileReader|FileInputStream|FileOutputStream)
```

### How Reproducible

Always (Default)

### Steps To Reproduce

1. Run analysis against [book-server](https://github.com/ibraginsky/book-server) app using the following rule:

```
  when:
    java.referenced:
      pattern: java.io.(File|FileWriter|FileReader|FileInputStream|FileOutputStream)
```

### Environment

```markdown
- **Analyzer Image SHA** latest 
- **JDTLS  base image SHA** latest
- **Provider Settings JSON**
```

**Comment by pranavgaikwad** [comment]
Something looks wrong about the classpath of the app which is creating an INACCURATE match that eventually gets filtered as this is a default search. When we try this same rule with the demo app (examples/java), it works. In that case, it generates an ACCURATE match. This rules out the possibility that the basic regex pattern `(X|Y|Z)` is broken.  Our initial investigation suggests that we will do more harm than good trying to fix this one-off. I am reducing the severity of this issue, and taking it out of beta.2. cc @shawn-hurley
